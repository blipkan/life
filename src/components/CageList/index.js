import React from 'react'
import * as PropTypes from 'prop-types'
import classnames from 'classnames'

import CageItem from '../CageItem'
import './index.scss'

const CageList = props => {
  const { items = [], hasScrollBar } = props

  return (
    <div className={classnames('cage-list', { scrolled: hasScrollBar })}>
      {items.map((item, index) => (
        <CageItem worldItem={item} key={item.id} index={index} />
      ))}
    </div>
  )
}

CageList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({})),
  hasScrollBar: PropTypes.bool
}

export default CageList
