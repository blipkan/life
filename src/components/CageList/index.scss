.cage-list {
  display: flex;
  padding: 0 10px 0 16px;
  flex-flow: column;
  width: auto;

  &.scrolled {
    padding-right: 4px;
  }
}
