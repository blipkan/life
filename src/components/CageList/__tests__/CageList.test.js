/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import CageList from '../index'

describe('CageList', () => {
  it('render component', () => {
    const component = shallow(<CageList />)
    expect(component).toMatchSnapshot()
  })
})
