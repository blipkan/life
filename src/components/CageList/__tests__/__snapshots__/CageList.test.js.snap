// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CageList render component 1`] = `
<div
  className="cage-list"
/>
`;
